lib/src/phy/channel/channel.cc

```diff
@@ -94,6 +100,25 @@ channel::channel(const channel::args_t& channel_args, uint32_t _nof_channels, sr
     srsran_channel_rlf_init(rlf, channel_args.rlf_t_on_ms, channel_args.rlf_t_off_ms);
   }
 
+  // Create Tuner
+  if (channel_args.tuner_enable && ret == SRSRAN_SUCCESS) {
+    tuner = std::make_unique<srsran_channel_tuner_t>(logger);
+    // this->tuner->sock.open(channel_args.tuner_name);
+    tuner->tuner_monitor_thread = std::make_unique<std::thread>([this, &logger]{
+      float new_gain;
+      do {
+        this->tuner->sock >> new_gain;
+        if (this->tuner->sock) {
+          this->tuner->tuner_attenuation.store(new_gain, std::memory_order_relaxed);
+          logger.info("Attenuation changed to {}", new_gain);
+        } else {
+          this->tuner->sock.clear();
```

Comment:
If you want to see how the program is running you can add log here too saying "corrupted input" or something... just dump as much as possible to log so you can analyze how the program is running.